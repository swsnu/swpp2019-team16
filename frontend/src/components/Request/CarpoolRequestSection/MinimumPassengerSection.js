import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Checkbox from '../../common/Checkbox';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  title: {
    marginBottom: '30px',
  },
  content: {
    marginBottom: '25px',
  },
  map: {
    width: '100%',
    flex: 1,
    marginBottom: '25px',
  },
  noLocationSelected: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '400px',
  },
});

MinimumPassengerSection.propTypes = {
  minimumPassengerOptions: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectedMinimumPassenger: PropTypes.string,
  onClickMinimumPassenger: PropTypes.func.isRequired,
};

function MinimumPassengerSection({
  minimumPassengerOptions,
  selectedMinimumPassenger,
  onClickMinimumPassenger,
}) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>How many co-passengers at least?</Typography>
      </div>
      <div className={styles.content}>
        {minimumPassengerOptions &&
          minimumPassengerOptions.map(minimumPassenger => (
            <Checkbox
              key={minimumPassenger}
              name={'minimumPassenger'}
              value={minimumPassenger}
              onClick={() =>
                selectedMinimumPassenger === minimumPassenger
                  ? onClickMinimumPassenger(null)
                  : onClickMinimumPassenger(minimumPassenger)
              }
              checked={minimumPassenger === selectedMinimumPassenger}
            />
          ))}
      </div>
    </div>
  );
}

export default MinimumPassengerSection;
