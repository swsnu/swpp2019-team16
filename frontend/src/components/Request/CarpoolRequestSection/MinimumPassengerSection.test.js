import React from 'react';
import { render } from '@testing-library/react';
import MinimumPassengerSection from './MinimumPassengerSection';

describe('<MinimumPassengerSection />', () => {
  it('should match with snapshot - default', async () => {
    const { container } = render(
      <MinimumPassengerSection
        minimumPassengerOptions={['2', '3', '4']}
        selectedMinimumPassenger={'2'}
        onClickMinimumPassenger={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot - when no selected option', async () => {
    const { container } = render(
      <MinimumPassengerSection
        minimumPassengerOptions={['2', '3', '4']}
        selectedMinimumPassenger={null}
        onClickMinimumPassenger={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
