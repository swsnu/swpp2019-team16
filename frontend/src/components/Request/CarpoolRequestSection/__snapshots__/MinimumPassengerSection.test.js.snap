// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MinimumPassengerSection /> should match with snapshot - default 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        How many co-passengers at least?
      </h4>
    </div>
    <div
      class="makeStyles-content-3"
    >
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="2"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-49 Mui-checked MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            2
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="3"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="3"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            3
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="4"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-48 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-51"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="4"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            4
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MinimumPassengerSection /> should match with snapshot - when no selected option 1`] = `
<div>
  <div
    class="makeStyles-root-83"
  >
    <div
      class="makeStyles-title-84"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        How many co-passengers at least?
      </h4>
    </div>
    <div
      class="makeStyles-content-85"
    >
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="2"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-130 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-133"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="2"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            2
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="3"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-130 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-133"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="3"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            3
          </span>
        </label>
      </div>
      <div>
        <label
          class="MuiFormControlLabel-root"
          data-testid="4"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-130 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-133"
                data-indeterminate="false"
                name="minimumPassenger"
                type="checkbox"
                value="4"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            4
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
